pkg/daemon/ceph/client/upgrade.go

```diff
@@ -104,6 +147,183 @@ func EnableNautilusOSD(context *clusterd.Context) error {
 		return fmt.Errorf("failed to disallow pre-nautilus osds and enable all new nautilus-only functionality: %+v", err)
 	}
 	logger.Infof("successfully disallowed pre-nautilus osds and enabled all new nautilus-only functionality")
+	return nil
+}
+
+// OkToStopOrContinue determines wether it's ok to stop or continue an upgrade
+func OkToStopOrContinue(context *clusterd.Context, namespace, deployment, action string, cephVersion cephver.CephVersion) error {
+	daemonName, err := findDaemonName(deployment)
+	if err != nil {
+		return fmt.Errorf("%+v", err)
```

Comment:
How do we tell the difference between returning an error and knowing that it is ok to stop or continue? For example, if we return an error here, will the caller retry until the timeout even though we will always fail? We might need to return a `(bool, error)` from the method to differentiate between errors and whether ok to stop.

Reply:
I have decided to not error but return nil and log a warning instead. If we can determine the daemon name that's probably for a good reason so there is nothing to do.